app/src/main/java/io/gnosis/safe/Tracker.kt

```diff
@@ -0,0 +1,60 @@
+package io.gnosis.safe
+
+import android.content.Context
+import android.os.Bundle
+import com.google.firebase.analytics.FirebaseAnalytics
+import com.google.firebase.crashlytics.FirebaseCrashlytics
+import io.gnosis.safe.di.ApplicationContext
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class Tracker @Inject constructor(@ApplicationContext context: Context) {
+
+    private val firebaseAnalytics: FirebaseAnalytics = FirebaseAnalytics.getInstance(context)
+
+    fun setNumSafes(numSafes: Int) {
+        firebaseAnalytics.setUserProperty(Param.NUM_SAFES, numSafes.toString())
+    }
+
+    fun setPushInfo(enabled: Boolean) {
+        firebaseAnalytics.setUserProperty(Param.PUSH_INFO, if (enabled) ParamValues.PUSH_ENABLED else ParamValues.PUSH_DISABLED)
+    }
+
+    private fun logEvent(name: String, attrs: Map<String, Any?>?) {
+        try {
+            val bundle = Bundle()
+            attrs?.let {
+                for ((key, value) in attrs) {
+                    when (value) {
+                        is String -> bundle.putString(key, value)
+                        is Long -> bundle.putLong(key, value)
+                        is Int -> bundle.putInt(key, value)
+                        is Number -> bundle.putFloat(key, value.toFloat())
+                    }
+                }
+            }
+            firebaseAnalytics.logEvent(name, bundle)
+        } catch (e: Exception) {
+            logException(e)
+        }
+    }
+
+    private fun logException(exception: Exception) {
+        FirebaseCrashlytics.getInstance().recordException(exception)
+    }
+
+    object Event {
+        // Put event names here
+    }
+
+    object Param {
+        val NUM_SAFES = "num_safes"
+        val PUSH_INFO = "push_info"
+    }
+
+    object ParamValues {
+        val PUSH_ENABLED = "enabled"
+        val PUSH_DISABLED = "disabled"
+    }
```

Comment:
I think for now it's ok, but we could provide something with callbacks whereby we build the event and just submit it leveraging polymorphism. Maybe it's overkill too xD. But it would be nice to have something like: 
```suggestion
    } object PUSH_INFO: Trackable {
       override val name = "push_info"
       //...
    }
    
   interface Trackable {
   val name: String
   }
```